atd-etl/cris_import/cris_import.py

```diff
@@ -225,48 +274,94 @@ def specify_extract_location():
     return zip_tmpdir
 
 
-def download_archives():
-    """
-    Connect to the SFTP endpoint which receives archives from CRIS and
-    download them into a temporary directory.
+def download_s3_archive():
+    session = boto3.Session(
+        aws_access_key_id=AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+    )
 
-    Returns path of temporary directory as a string
-    """
+    s3 = session.client("s3")
+    bucket = S3_EXTRACT_BUCKET
+    prefix = f"{DEPLOYMENT_ENVIRONMENT}/database/"
+
+    # Check if the SQLite DB exists in the specified directory on S3
+    db_key = f"{prefix}uploads.sqlite"
+    try:
+        s3.head_object(Bucket=bucket, Key=db_key)
+    except botocore.exceptions.ClientError as e:
+        # If the DB does not exist, copy it from the default directory on S3
+        if e.response["Error"]["Code"] == "404":
+            default_db_key = "default_database/uploads.sqlite"
+            copy_source = {"Bucket": bucket, "Key": default_db_key}
+            s3.copy(copy_source, bucket, db_key)
+
+    uploads_prefix = f"{DEPLOYMENT_ENVIRONMENT}/uploads/"
+    # Get list of all objects in the bucket with the specified prefix
+    objects = s3.list_objects(Bucket=bucket, Prefix=uploads_prefix)
```

Comment:
this is going to run into the limitation that you must paginate through the object list beyond the first thousand results, but i would not be concerned about this if we were moving processed extracts to a separate subdirectory.

Reply:
We spoke about this, and with the lifecycle rules in place, we were in a good place re: 1000 entries at a time, but we're in a doubly good place now because the script has been updated to move the files into a processed folder too.